tekton/utils/pipeline_run_builder.go

```diff
@@ -204,8 +228,12 @@ func (b *PipelineRunBuilder) WithServiceAccount(serviceAccount string) *Pipeline
 }
 
 // WithTimeouts sets the Timeouts for the PipelineRun.
-func (b *PipelineRunBuilder) WithTimeouts(timeouts *tektonv1.TimeoutFields) *PipelineRunBuilder {
-	b.pipelineRun.Spec.Timeouts = timeouts
+func (b *PipelineRunBuilder) WithTimeouts(timeouts, defaultTimeouts *tektonv1.TimeoutFields) *PipelineRunBuilder {
+	if timeouts == nil || *timeouts == (tektonv1.TimeoutFields{}) {
+		b.pipelineRun.Spec.Timeouts = defaultTimeouts
```

Comment:
This still works if `defaultTimeouts` are not defined in the RSC, right?

Reply:
Yes, it should as it will set it to nil which was the previous behavior.